packages/terra-application-utility/src/header/_HeaderUtilityMenu.jsx

```diff
@@ -0,0 +1,256 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames/bind';
+import 'terra-base/lib/baseStyles';
+import ContentContainer from 'terra-content-container';
+import IconLeft from 'terra-icon/lib/icon/IconLeft';
+import MenuDivider from '../_UtilityMenuDivider';
+import Utils from '../_Utils';
+import MenuItem from './_HeaderUtilityMenuItem';
+import styles from './_HeaderUtilityMenu.scss';
+
+const cx = classNames.bind(styles);
+
+const propTypes = {
+  /**
+   * Key of the top level menu.
+   */
+  initialSelectedKey: PropTypes.string.isRequired,
+  /**
+   * Indicates if the height is bound to a value.
+   */
+  isHeightBounded: PropTypes.bool,
+  /**
+   * The config file containing the menu items to be rendered.
+   */
+  menuItems: PropTypes.arrayOf((Utils.itemShape)).isRequired,
+  /**
+   * The function to trigger when a menu item is selected.
+   */
+  onChange: PropTypes.func.isRequired,
+  /**
+   * The function that closes the menu.
+   */
+  onRequestClose: PropTypes.func,
```

Comment:
Does `onRequestClose` need to be required?

Reply:
No, the `onRequestClose` prop is cloned onto the menu instead of being passed in at component creation. React will throw a warning if this prop is required.

Reply:
We should probably add a note to indicate this is provided then.